### PR TITLE
Update environments to virtualMachines resource type.

### PR DESCRIPTION
**Pull request by shichaoh**

### CODE REVIEW

File: ARMTemplates/101-dtl-create-vm-builtin-user/azuredeploy.json [hunk 1]
```diff
@@ -29,26 +29,21 @@
...
-            "name": "[parameters('newVMName')]",
-            "vmTemplateName": "[parameters('existingCustomImageName')]",
-            "size": "[parameters('newVMSize')]",
-            "isAuthenticationWithSshKey": false,
-            "sshKey": "",
-            "userName": "",
-            "password": ""
-          }
-        ]
+        "vmTemplateName": "[parameters('existingCustomImageName')]",
```
Comment (mithunshanbhag) [inline ARMTemplates/101-dtl-create-vm-builtin-user/azuredeploy.json]: Shouldn't there be another property for the VM name?

*"name": "[parameters('newVMName')]"*

File: ARMTemplates/101-dtl-create-vm-username-pwd-customimage/azuredeploy.json [hunk 2]
```diff
@@ -41,26 +41,21 @@
...
-            "name": "[parameters('newVMName')]",
-            "vmTemplateName": "[parameters('existingCustomImageName')]",
-            "size": "[parameters('newVMSize')]",
-            "isAuthenticationWithSshKey": false,
-            "userName": "[parameters('userName')]",
-            "sshKey": "",
-            "password": "[parameters('password')]"
-          }
-        ]
+        "vmTemplateName": "[parameters('existingCustomImageName')]",
```
Comment (mithunshanbhag) [inline ARMTemplates/101-dtl-create-vm-username-pwd-customimage/azuredeploy.json]: Same as above. Shouldn't there be another property for the VM name?

File: ARMTemplates/101-dtl-create-vm-username-pwd-galleryimage/azuredeploy.json [hunk 3]
```diff
@@ -1,104 +1,99 @@
...
+      "metadata": {
+        "description": "The name of the new vm to be created."
+      }
     },
-    "variables": {
-        "resourceName": "[concat(parameters('existingLabName'), '/', parameters('newVMName'))]",
-        "resourceType": "Microsoft.DevTestLab/labs/environments"
+    "existingLabName": {
+      "type": "string",
+      "metadata": {
```
Comment (mithunshanbhag) [inline ARMTemplates/101-dtl-create-vm-username-pwd-galleryimage/azuredeploy.json]: Looks like something is wrong with this file. Was the metadata section wrongly placed here?

File: ARMTemplates/101-dtl-create-vm-username-ssh-customimage/azuredeploy.json [hunk 4]
```diff
@@ -41,26 +41,21 @@
...
-            "name": "[parameters('newVMName')]",
-            "vmTemplateName": "[parameters('existingCustomImageName')]",
-            "size": "[parameters('newVMSize')]",
-            "isAuthenticationWithSshKey": true,
-            "userName": "[parameters('userName')]",
-            "sshKey": "[parameters('sshKey')]",
-            "password": ""
-          }
-        ]
+        "vmTemplateName": "[parameters('existingCustomImageName')]",
```
Comment (mithunshanbhag) [inline ARMTemplates/101-dtl-create-vm-username-ssh-customimage/azuredeploy.json]: Same as above. Shouldn't there be another property for the VM name?

File: ARMTemplates/101-dtl-create-vm-builtin-user/azuredeploy.json (hunk 1 above)
Comment (shichaoh) [inline ARMTemplates/101-dtl-create-vm-builtin-user/azuredeploy.json]: nope. the name is specific in the "name" out side the properties. basically using the same name as the DTL VM resource name.

File: ARMTemplates/101-dtl-create-vm-username-pwd-galleryimage/azuredeploy.json (hunk 3 above)
Comment (shichaoh) [inline ARMTemplates/101-dtl-create-vm-username-pwd-galleryimage/azuredeploy.json]: bad diff in GitHub, actually, just some additional changes about indention.

**Comment by mithunshanbhag** [comment]
LGTM

**Pull request merged**